plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeColumnHandle.java

```diff
@@ -75,14 +81,19 @@ public DeltaLakeColumnHandle(
             @JsonProperty("fieldId") OptionalInt fieldId,
             @JsonProperty("physicalName") String physicalName,
             @JsonProperty("physicalType") Type physicalType,
-            @JsonProperty("columnType") DeltaLakeColumnType columnType)
+            @JsonProperty("columnType") DeltaLakeColumnType columnType,
+            @JsonProperty("baseColumnIndex") int baseColumnIndex,
+            @JsonProperty("projectionInfo") Optional<DeltaLakeColumnProjectionInfo> projectionInfo)
     {
         this.baseColumnName = requireNonNull(baseColumnName, "baseColumnName is null");
         this.baseType = requireNonNull(baseType, "baseType is null");
         this.fieldId = requireNonNull(fieldId, "fieldId is null");
         this.physicalName = requireNonNull(physicalName, "physicalName is null");
         this.physicalType = requireNonNull(physicalType, "physicalType is null");
         this.columnType = requireNonNull(columnType, "columnType is null");
+        checkArgument(baseColumnIndex >= 0 || columnType == PARTITION_KEY || columnType == SYNTHESIZED, "baseColumnIndex is negative");
```

Comment:
what is the expected value for `baseColumnIndex` for partition and synthesized columns?
enforce in the constructor

Reply:
Not needed now. Removed `baseColumnIndex`.

https://github.com/trinodb/trino/pull/17085#discussion_r1173565261